controlplane/eks/api/v1beta2/types_class.go

```diff
@@ -0,0 +1,180 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta2
+
+import (
+	infrav1 "sigs.k8s.io/cluster-api-provider-aws/v2/api/v1beta2"
+)
+
+// AWSManagedControlPlaneClassSpec defines the AWSManagedControlPlane properties that may be shared across several AWS managed control planes.
+type AWSManagedControlPlaneClassSpec struct {
+	// MachineTemplate contains information about how machines
+	// should be shaped when creating or updating a control plane.
+	// For the AWSManagedControlPlaneTemplate, this field is used
+	// only to fulfill the CAPI contract.
+	// +optional
+	MachineTemplate *AWSManagedControlPlaneTemplateMachineTemplate `json:"machineTemplate,omitempty"`
```

Comment:
I'm not sure why this is necessary? Can you link to the CAPI contract that nees to be fulfilled that requires this to be added?

Reply:
It's part of the control plane provider contract: https://cluster-api.sigs.k8s.io/developer/providers/contracts/control-plane#controlplane-machines.
CAPZ has the machine templates, I’m totally fine not having the machine template field here.

Reply:
It's not required here because the control plane nodes are not represented by `Machines`, as the note says:

> In case you are developing a control plane provider which uses a Cluster API Machine object to represent each control plane instance, following fields MUST be implemented in the ControlPlane spec.